src/modules/Channel/components/SuggestedMentionList/index.tsx

```diff
@@ -56,7 +56,24 @@ function SuggestedMentionList(props: SuggestedMentionListProps): JSX.Element {
   const [searchString, setSearchString] = useState('');
   const [lastSearchString, setLastSearchString] = useState('');
   const [currentUser, setCurrentUser] = useState<User>(null);
-  const [currentMemberList, setCurrentMemberList] = useState<Array<Member>>([]);
+  const [currentMemberList, setCurrentMemberList] = useState<Member[]>([]);
+
+  // We have to be specific like this or React would not recognize the changes in instances.
+  const currentSearchedMemberList = [
+    channelInstance.members.map((member: Member) => member.userId).join(),
+    channelInstance.members.map((member: Member) => member.nickname).join(),
+    channelInstance.members.map((member: Member) => member.isActive).join(),
+  ];
+
+  const getCurrentSearchedMemberList = (): Member[] => useMemo(() => {
+    return currentChannel.members
+      .sort((a, b) => a.nickname?.localeCompare(b.nickname))
+      .filter(
+        (member) => member.nickname?.toLowerCase().startsWith(searchString.toLowerCase())
+        && member.userId !== currentUser?.userId
+        && member.isActive,
+      ).slice(0, maxSuggestionCount);
+  }, currentSearchedMemberList);
```

Comment:
It seems like calculations and memoization are being performed even when they are not necessary.
It appears to be a better approach to create a separate function and generate the list only when needed by calling it.

Reply:
Can we achieve that by replacing useMemo with useCallback?

Reply:
It looks like it can be made into a pure function.

Reply:
I see. ill put back to pure function